### PR TITLE
[Feature][Meetings][1738002]

### PR DESCRIPTION
**Pull request by v-rodriguezd**
Added Meeting Details API and meeting event details. Please review.

### CODE REVIEW

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 1]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+## Real-time Teams meeting events
+
+The user can receive real-time meeting events. As soon as any app is associated with a meeting, the actual meeting start and meeting end time are shared with the bot.
+
+Actual start and end time of a meeting are different from the scheduled start and end time. The meeting details API and real-time start and end time meeting events are made available.
+
+There are two major categories of events that are supported:
+
+* Meeting start and end event
+* Participant update event
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:310]: Remove "Participant update event". We don't support that

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 2]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+        "tenantId": "<Tenant ID>" 
+    }
+} 
+```
+
+## Real-time Teams meeting events
+
+The user can receive real-time meeting events. As soon as any app is associated with a meeting, the actual meeting start and meeting end time are shared with the bot.
+
+Actual start and end time of a meeting are different from the scheduled start and end time. The meeting details API and real-time start and end time meeting events are made available.
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:305]: Reword: 
Actual start and end time of a meeting are different from the scheduled start and end time. The meeting details API exposes the scheduled start and end times while the event gives the actual start and end time

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 3]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+        "source": null,
+        "meeting": {
+            "id": "MCMxOTptZWV0aW5nX05UTTNaREpqT1RVdFpHUmhPUzAwTXpZeExUazVOREF0TXpZNE0ySXpaV0ZqWkdFMUB0aHJlYWQudjIjMA=="
+        }
+    },
+    "value": {
+        "MeetingType": "Scheduled",
+        "Title": "Meeting Start/End Event in Canary",
+        "Id":
+"19:meeting_NTM3ZDJjOTUtZGRhOS00MzYxLTk5NDAtMzY4M2IzZWFjZGE1@thread.v2",
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:358]: Indent properly

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 4]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+        "JoinUrl": "https://teams.microsoft.com/l/meetup-join/19%3ameeting_NTM3ZDJjOTUtZGRhOS00MzYxLTk5NDAtMzY4M2IzZWFjZGE1%40thread.v2/0?context=%7b%22Tid%22%3a%22e15762ef-a8d8-416b-871c-25516354f1fe%22%2c%22Oid%22%3a%22f30ba569-abef-4e97-8762-35f85cbae706%22%7d",
+        "StartTime": "2021-04-29T16:10:41.1252256Z"
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:360]: Indent properly

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md (hunk 3 above)
Comment (rarang) [reply msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:358]: Can you please replace the start event with the payload below? This is what I shared in the doc as well @SurbhiJainMSFT 
image

https://microsoft-my.sharepoint-df.com/:w:/p/rarang/EfIWi3sOGAhLvrjF1gf1ifcBZiSNrrQ4YGqb3_8UqsnTWA?e=8hzRjU

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 5]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+    },
+    "locale": "en-US"
+}
+```
+
+### Example of meeting end event payload
+
+The following code provides an example of meeting end event payload:
+
+```json
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:370]: Replace meeting end event with this. This is in the word doc

image

**Review by rarang — CHANGES_REQUESTED** [review]
Please remove "participant join" as an event. We don't support it
Also, please update the meeting start and end payloads with what i have

**Review by rarang — APPROVED** [review]

**Review by rarang — APPROVED** [review]

**Review by rarang — DISMISSED** [review]

**Review by rarang — APPROVED** [review]

**Review by v-abirade — APPROVED** [review]
Approved

**Review by surbhigupta12 — APPROVED** [review]
Approved.

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 6]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+
+Actual start and end time of a meeting are different from the scheduled start and end time. The meeting details API provides the scheduled start and end time while the event provides the actual start and end time.
+
+### Example of meeting start event payload
+
+The following code provides an example of meeting start event payload:
+
+```json
+{ 
+    "name": "Microsoft/MeetingStart",
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:319]: **replace** "Microsoft/MeetingStart" **with** "application/vnd.microsoft.meetingStart"

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 7]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+{ 
+    "name": "Microsoft/MeetingStart", 
+    "type": "event", 
+    "timestamp": "2021-04-29T16:10:41.1252256Z", 
+    "id": "123", 
+    "channelId": "msteams", 
+    "serviceUrl": "https://microsoft.com", 
+    "from": { 
+        "id": "userID", 
+        "name": "",
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:327]: remove the field "name":"",

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 8]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+}
+```
+
+### Example of meeting end event payload
+
+The following code provides an example of meeting end event payload:
+
+```json
+{ 
+    "name": "Microsoft/MeetingEnd",
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:372]: **replace**  "Microsoft/MeetingEnd" **with** "application/vnd.microsoft.meetingEnd"

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 9]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+{ 
+    "name": "Microsoft/MeetingEnd", 
+    "type": "event", 
+    "timestamp": "2021-04-29T16:17:17.4388966Z", 
+    "id": "123", 
+    "channelId": "msteams", 
+    "serviceUrl": "https://microsoft.com", 
+    "from": { 
+        "id": "user id", 
+        "name": "",
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:380]: remove the field "name":"",

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md [hunk 10]
```diff
@@ -220,6 +227,234 @@ POST /v3/conversations/{conversationId}/activities
...
+
+To deserialize the json payload, a model object is introduced to get the metadata of a meeting. The metadata of a meeting resides in the `value` property in the event payload. The `MeetingStartEndEventvalue` model object is created, whose member variables correspond to the keys under the `value` property in the event payload.
+
+The following code shows how to capture the metadata of a meeting that is `MeetingType`, `Title`, `Id`, `JoinUrl`, `StartTime`, and `EndTime` from a meeting start and end event:
+
+```csharp
+protected override async Task OnEventActivityAsync(
+ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+{
+    // Event Name is either `Microsoft/MeetingStart` or `Microsoft/MeetingEnd`
```
Comment (rarang) [inline msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:431]: replace the comment with the following 

Event Name is either 'application/vnd.microsoft.meetingStart' or 'application/vnd.microsoft.meetingEnd'

File: msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md (hunk 6 above)
Comment (v-rodriguezd) [reply msteams-platform/apps-in-teams-meetings/create-apps-for-teams-meetings.md:319]: @rarang cannot find these in the create-apps-for-teams-meetings.md file do we need to add Meeting Details API first as the section was removed as per your request.

**Comment by v-rodriguezd** [comment]
@rarang cannot find these in the create-apps-for-teams-meetings.md file do we need to add Meeting Details API first as the section was removed as per your request.

**Pull request merged**